Objects/typeobject.c

```diff
@@ -3444,15 +3499,70 @@ PyType_FromMetaclass(PyTypeObject *metaclass, PyObject *module,
                     vectorcalloffset = memb->offset;
                 }
             }
+            break;
+        case Py_tp_doc:
```

Comment:
We should consider refactoring out some more helpers; for example this case. Maybe save that for another PR.